presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayNormalizeFunction.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.type.DoubleType;
+import com.facebook.presto.common.type.RealType;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlNullable;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.TypeParameter;
+import com.facebook.presto.spi.function.TypeParameterSpecialization;
+
+import static com.facebook.presto.spi.StandardErrorCode.FUNCTION_IMPLEMENTATION_MISSING;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.util.Failures.checkCondition;
+import static java.lang.String.format;
+
+@ScalarFunction("array_normalize")
+@Description("Normalizes an array by dividing each element by the p-norm of the array.")
+public final class ArrayNormalizeFunction
+{
+    private static final ValueAccessor doubleValueAccessor = new DoubleValueAccessor();
```

Comment:
Updated the code to create static members to address the efficiency issue you mentioned. Please take another look at this file. Thanks.

Reply:
This is much cleaner! One more nit: for `static final` constants, we use capital naming convention:
- `DOUBLE_VALUE_ACCESSOR`
- `REAL_VALUE_ACCESSOR`

Reply:
Done. Thanks.